### PR TITLE
Do not remove server constants from GROUP BY key for secondary query.

### LINKED ISSUE #62682: "Not found column [..] in block" with clusterAllReplicas
This doesn't work:
```
:) select serverUUID() as s, sum(dummy) from clusterAllReplicas(both, system.one) group by s

[...]
Received exception from server (version 24.4.1):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column serverUUID() in block. There are only columns: sum(__table1.dummy): While executing Remote. (NOT_FOUND_COLUMN_IN_BLOCK)
```

But this works:
```
:) select cityHash64(serverUUID()) + dummy as s, sum(dummy) from clusterAllReplicas(both, system.one) group by s

[...]
   ┌───────────────────s─┬─sum(dummy)─┐
1. │ 6220104575184213473 │          0 │
2. │ 7064952209212228648 │          0 │
   └─────────────────────┴────────────┘

2 rows in set. Elapsed: 0.006 sec. 
```

The cluster is just 2 servers running on localhost:
```
remote_servers:                                                                                                                                                                                                                        
  both:                                                                                                                                                                                                                                
    shard:                                                                                                                                                                                                                             
      replica:                                                                                                                                                                                                                         
        host: 127.0.0.1                                                                                                                                                                                                                
        port: 9000                                                                                                                                                                                                                     
      replica:                                                                                                                                                                                                                         
        host: 127.0.0.3                                                                                                                                                                                                                
        port: 9010                                                                                                                                                                                                                     
```

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `Not found column in block` error for distributed queries with server-side constants in `GROUP BY` key. Fixes #62682

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 48213c2c3ef2ac4cc9c06e3f76268862a5ba0859 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63047/48213c2c3ef2ac4cc9c06e3f76268862a5ba0859/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Planner/PlannerExpressionAnalysis.cpp [hunk 1]
```diff
@@ -100,7 +102,7 @@ std::optional<AggregationAnalysisResult> analyzeAggregation(const QueryTreeNodeP
                     auto is_constant_key = grouping_set_key_node->as<ConstantNode>() != nullptr;
                     group_by_with_constant_keys |= is_constant_key;
 
-                    if (is_constant_key && !aggregates_descriptions.empty())
+                    if (!is_secondary_query && is_constant_key && !aggregates_descriptions.empty())
```
Comment (alexey-milovidov) [inline src/Planner/PlannerExpressionAnalysis.cpp:105]: Missing a comment.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by tavplubix** [comment]
@alexey-milovidov, @KochetovNicolai, CI failures were related, it breaks `02967_analyzer_fuzz`

**Comment by Algunenano** [comment]
> @alexey-milovidov, @KochetovNicolai, CI failures were related, it breaks 02967_analyzer_fuzz

Reproduces with:

```
SELECT
    2147483647,
    count(pow(NULL,  1.0001))
FROM remote(test_cluster_two_shards,  system,  one)
GROUP BY
    makeDateTime64(NULL,  NULL,  pow(NULL,  '257') - '-1',  '0.2147483647',  257),
    makeDateTime64(pow(pow(NULL,  '21474836.46') - '0.0000065535',  1048577),  '922337203685477580.6',  NULL,  NULL,  pow(NULL,  1.0001) - 65536,  NULL)
WITH CUBE
SETTINGS
    allow_experimental_analyzer = 1,
    group_by_two_level_threshold=1,
    group_by_two_level_threshold_bytes=33950592;
```

This breaks CI, so we need to fix it or revert

**Comment by Algunenano** [comment]
Minimized:
```
SELECT count()
FROM remote(test_cluster_two_shards, system, one)
GROUP BY 'hi'
SETTINGS allow_experimental_analyzer = 1, group_by_two_level_threshold = 1, group_by_two_level_threshold_bytes = 33950592
```

`DB::Exception: Unknown aggregated data variant.: While executing GroupingAggregatedTransform. (UNKNOWN_AGGREGATED_DATA_VARIANT)`

**Pull request merged**